### DISCUSSION TITLE
Admin Update

### DISCUSSION THREAD

**Discussion by kasperkristensen** [medusajs/medusa#5954] [RFC]
Earlier this year, we introduced [Admin Extensions](https://medusajs.com/blog/medusa-ui-and-general-availability-of-admin-extensions/), allowing developers to customize the Medusa Admin to meet their specific needs. It has been exciting to see the community’s use of this feature, but it is obvious to us that there is room for improving the developer experience.

We've recently begun work on improving how Admin extensions and the Admin in general function. We aim to roll out some of these improvements in the first quarter of 2024. Below is a summary of our current projects and future plans. 

As always, we welcome your feedback, so please don't hesitate to share your suggestions or questions.

### Improved Developer Experience

One thing that has been apparent is that the development experience when working with Admin extensions has been less than ideal. This has mainly been due to the lack of proper HMR, which resulted in each change taking several seconds to take effect. To provide a better DX, we are migrating our Admin build tooling to Vite and moving the extension mechanism into a new Vite plugin.

The new tooling delivers almost instant updates when building extensions and offers an improved development server and faster build times. Below is a video demonstrating the power of HMR with Vite.

https://github.com/medusajs/medusa/assets/45367945/642907c4-08aa-427b-b89f-045f1437e74b

We aim to ensure a seamless migration. No changes will be needed for extensions that are part of a Medusa backend project. However, for extensions included in a plugin, a new version bundled with our new build tool will need to be released for the extensions to be picked up. We will publish more information and a migration guide as we approach the initial beta testing.

### Admin Revamp

A few months ago, we launched `@medusajs/ui`, a React component library for our new design system. We've begun migrating our Admin to this new design system and anticipate delivering a completely updated admin experience in the near future.

This update will ensure a consistent aesthetic across the entire application and will also give us an opportunity to enhance the UX. As part of our UI revamp, we're committed to ensuring responsiveness so the application functions efficiently whether the user is on a phone, tablet, or computer.

Additionally, we're also investing resources into cleaning up technical debt and addressing bugs. By focusing on these areas, we aim to ensure the stability and reliability of the Admin and improve the overall experience for all users.

## Future Roadmap

Having outlined our plans for the coming months, we want to share a brief comment on what else is in store for Medusa Admin next year.

### **Extension Localization**

Localization was recently added to the Admin. It's been wonderful to see the numerous contributions from the community, helping us translate the Admin into more than 20 different languages. We aim to enhance this by also making UI extensions translatable.

### Extending Existing Admin Components

We've seen quite a few requests for extending existing parts of the Admin, such as the product creation form. This feature is on our roadmap for future support, though we're not ready to begin work on it just yet. As we approach Medusa 2.0, we'll be able to share more details about how and when we'll support full extensibility of the Admin.

**Comment by alightinastorm** [comment]
Hoping that #4750 and #5679 will be considered when rebuilding the admin..

**Reply by kasperkristensen** [reply]
Regarding #4750 that is what I am alluding to under "Extending Existing Admin Components". It is something we want to add support for, but as we are working on Medusa 2.0, how you customize entities in Medusa is changing, and we want to land those changes before we start working on supporting it in the Admin. But it is on the roadmap, and it is something we will be picking up later next year. 

I like the idea of having a setting for decimal and group separators. The new components in `@medusajs/ui` all use the users locale to determine which separators to use. But we could add some setting to override this globally. What would the ideal solution be in your opinion? Is it a setting that applies to the individual admin user, or something that is stored on the store entity and applies to all users?

**Reply by alightinastorm** [reply]
number formatting should be tied to the locale, actually

also you guys seem to be moving (very) fast on breaking changes.. seeing that medusa 2.0 is probably a totally new way of building with medusa, i'm worried about launching our product on 1.x

**Reply by kasperkristensen** [reply]
Sorry let me be more clear, I am not talking about breaking existing customizations. 

But with Medusa 2.0, we will be introducing a new way of extending entities that does not require modifying the existing DB schema, and that is what we will be recommending that people use then. It is also what we plan on being the mechanism that we will rely on to know which fields to render in the different Admin UI forms. Initial thinking here is to have some form of way to read which fields exists, and the type of the fields, and then use that to render specific fields. An approach that might seem familiar if you have worked with CMS systems like Sanity. But we won't know the specifics until we get a bit closer to Medusa 2.0.

**Reply by octalpixel** [reply]
Excited for this! But would also be nice if there is documentation around building/modifying the existing admin panel without the extension. Right now the only best way would be to link it as an local module if I am correct.

**Comment by pmltechpile** [comment]
Would love to see [2768](https://github.com/medusajs/medusa/discussions/2768) in the next year

**Comment by chrislaai** [comment]
<img width="529" alt="image" src="https://github.com/medusajs/medusa/assets/134155599/acdb0bc7-508a-4479-b5bb-dc3a0b91a5d9">
<img width="48" alt="image" src="https://github.com/medusajs/medusa/assets/134155599/a10a8918-cae4-49f6-a949-884324ea5004">
I just played with the new Admin, and the purpose of this icon is not obvious, maybe the common one is easier to see and understand, and I don't know why it is needed on the desktop. Are there any benefits to hiding the sidebar on desktop? Keep it simple and intuitive🙏

**Comment by yazfield** [comment]
> Extending Existing Admin Components

Will there be a "hacky" way in medusa 2.0 to extend admin components before you have the time to work on this solution?

**Reply by moussaab-moulim** [reply]
what about extending from fields 
like adding a richtext editor for the description field in add product page
